### DISCUSSION TITLE
Feature/Modification: Make harder guest generation actually harder?

### DISCUSSION THREAD

**Discussion by Dracheron** [OpenRCT2/OpenRCT2#15739] [Feature requests]
This issue can make the harder guest generation actually harder if implemented correctly.

The following changes to the algorithm could make harder guest generation actually worth something (and could possibly be referred to as Master guest generation)

- The Soft Guest cap caps out at 800 instead of 1000 without any penalties
- After the soft guest cap of 800, in order for a ride to contribute 1/8 of their soft guest cap value, the ride must meet the requirements in question:

1. The ride has a minimum excitement of 3.00 except under low intensity conditions
2. The ride is either less than 7 years old OR over 25 years old
3. The ride intensity meets one of three conditions:

- In guests preferring higher intensities, the intensity CANNOT be less than 8.50
- In guests preferring lower intensities, the intensity CANNOT be higher than 5.00
- Under any circumstances:

1. Intensity is not higher than 9.95
2. Nausea is not higher than 7.50
3. Lateral G's CANNOT be over 2.50
4. Negative Vertical G's cannot be be over -2.15
5. Positive Vertical G's cannot be over 5.40
6. Excitement rating is not less than 1/3 of the Intensity rating
7. Excitement rating is not less than 1/2 of the Nausea rating
8. Cannot at any point have had a crash, even during testing 

- In order for a ride to contribute 1/4 of their soft guest cap value, in addition to meeting all of the requirements in the previous list, must meet the following requirements:

1. The ride is less than 3 years old
2. The ride has a minimum excitement of 4.50
3. The ride has a minimum length of 200m

- In order for a ride to contribute 1/2 of their soft guest cap value, in addition to meeting the requirements for 1/8 AND 1/4 contributions, must meet the following requirements:

1. The ride meets all minimum requirements for full stats
2. The ride is less than one year old
3. The ride has a minimum excitement rating of 6.50
4. The ride has a minimum length of 600m

As a note, this new maximum value after 800 overrides the normal value, so as such, the 800 soft cap will be determined first from rides ineligible for going over the soft cap that have not crashed (a ride that has crashed will always not contribute to the soft cap in any way, shape, or form), until the 800 soft cap has been reached, or the number of ineligible rides are exhausted, then the rides that contribute to the soft cap will come from the rides eligible for the 1/8 contribution (where the soft cap is temporarily suspended until the rides adding up to the 800 soft cap are determined), then the 1/4 eligible, then the 1/2 eligible (note that early on the soft cap would be so low that it doesn't come into effect initially).

Now the rides aspect would be just the first part of the changes.  The second part of this change would affect how advertising can increase guest generation, but it too will have a cap.  For advertising to have a good effect, the following will need to be done:
- Advertising has NO effect if a ride has crashed in the last 5 years
- Advertising has NO effect on a park with less than 600 park rating
- Advertising has NO effect if a fatality has occurred in the last 12 years
- Advertising is affected by rewards (where efficiency is added instead of multiplied, so having beneficial rewards is helpful)
- Advertising is affected proportionally from .1% at 600 park rating to 35% at 999 park rating
- Advertising gives a temporary soft cap of 25% above the normal soft cap, and is modifiable from 0% for park rating under 600 to 40% at maximum park rating

Hopefully this gives some ideas as to make actual harder guest generation.

**Comment by James103** [comment]
That would be better suited for a plugin, IMO.
